cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/ShuntConversion.java

```diff
@@ -43,31 +46,48 @@ public void convert() {
         int sections = getSections(p, normalSections);
         sections = Math.abs(sections);
         maximumSections = Math.max(maximumSections, sections);
-        double bPerSection = 0;
-        if (p.containsKey(CgmesNames.B_PER_SECTION)) {
-            bPerSection = p.asDouble(CgmesNames.B_PER_SECTION, 0.0);
-        } else {
+        ShuntCompensatorAdder adder = voltageLevel().newShuntCompensator().setCurrentSectionCount(sections);
+        String shuntType = p.getId("type");
+        if ("LinearShuntCompensator".equals(shuntType)) {
+            double bPerSection = p.asDouble(CgmesNames.B_PER_SECTION, Float.MIN_VALUE);
+            if (bPerSection == 0) {
+                double bPerSectionFixed = Double.MIN_VALUE;
+                fixed(CgmesNames.B_PER_SECTION, "Can not be zero", bPerSection, bPerSectionFixed);
+                bPerSection = bPerSectionFixed;
+            }
+            double gPerSection = p.asDouble("gPerSection", Double.NaN);
+            adder.newLinearModel()
+                    .setBPerSection(bPerSection)
+                    .setGPerSection(gPerSection)
+                    .setMaximumSectionCount(maximumSections)
+                    .add();
+        } else if ("NonlinearShuntCompensator".equals(shuntType)) {
+            ShuntCompensatorNonLinearModelAdder modelAdder = adder.newNonLinearModel();
             PropertyBags ss = context.cgmes().nonlinearShuntCompensatorPoints(id);
-            final int nlsections = sections;
-            double sumSections = ss.stream()
-                    .filter(s -> s.asInt("sectionNumber") <= nlsections)
-                    .map(s -> s.asDouble("b"))
-                    .reduce(0.0, Double::sum);
-            // Convert to a shunt compensator with a single section
-            maximumSections = 1;
-            sections = 1;
-            bPerSection = sumSections;
-        }
-        if (bPerSection == 0) {
-            float bPerSectionFixed = Float.MIN_VALUE;
-            fixed(CgmesNames.B_PER_SECTION, "Can not be zero", bPerSection, bPerSectionFixed);
-            bPerSection = bPerSectionFixed;
+            for (PropertyBag sp : ss) {
+                int sectionNumber = sp.asInt(SECTION_NUMBER);
+                double b = ss.stream().filter(s -> s.asInt(SECTION_NUMBER) <= sectionNumber).map(s -> s.asDouble("b")).reduce(0.0, Double::sum);
+                if (sectionNumber == 0) {
+                    if (b != 0) {
+                        fixed("Shunt.section0", "Susceptance of section 0 should always be zero (disconnected state)", b, 0.);
```

Comment:
Is it in the CGMES documentation that section 0 should have a b equal to zero ?

Reply:
It doesn't but I don't know if it makes sense? 
Actually I think it is linked to your other remark: to be "realistic", it doesn't really matter if `b` is 0 at section 0 as long as there is a section where `b` is 0 (a section where the shunt can be considered as disconnected). Maybe we should go in that sense? However in CGMES datasets, even in explicit section 0, the susceptance is always 0 from what I've seen.

Reply:
From Implementation guide:
````Use the cim:ShuntCompensator.sections to control the impedance where cim:ShuntCompensator.sections=0 means zero admittance. For linear shunt this means multiplying with the per section admittance results in zero impedance. For nonlinear shunts the cim:NonlinearShuntCompensatorPoint.sectionNumber=0 shall have the corresponding admittance values equal zero.````

Reply:
I don't understand the meaning of a zero admittance section. For me this should not be a requirement in IIDM but we need to accept it in order to keep the same number of section between CGMES and IIDM and then be able to update back the current section in case of regulation.
Why having a special meaning for section 0? section 0 is just the first one and could have an admittance. Also the shunt should accept 0 sections activated.

Reply:
In what I implemented, section 0 is in fact the disconnected state (like linear shunts: when `sectionIndex == 0 `, b and g are also at 0 because they are proportional to `sectionIndex` ; this is also the convention adopted in CGMES). We can not do things like this but this would imply:
- that a load-flow can disconnect the shunt's terminal **or** we create a parameter to indicate if the shunt is in disconnected state or not (and in which case, we do not consider its b and g)
- having different definitions of what section 0 is for linear and non linear shunts (that might be leading to different behaviors after a load-flow)